src/plugins/unified_search/public/saved_query_management/saved_query_management_list.tsx

```diff
@@ -112,41 +166,96 @@ const itemLabel = (attributes: SavedQueryAttributes) => {
+  }
+);
+
+const SAVED_QUERY_PAGE_SIZE = 5;
+const SAVED_QUERY_SEARCH_DEBOUNCE = 500;
+
+export const SavedQueryManagementList = ({
   showSaveQuery,
   loadedSavedQuery,
+  savedQueryService,
+  queryBarMenuRef,
   onLoad,
   onClearSavedQuery,
-  savedQueryService,
   onClose,
-  hasFiltersOrQuery,
-}: SavedQueryManagementListProps) {
-  const kibana = useKibana<IUnifiedSearchPluginServices>();
-  const [savedQueries, setSavedQueries] = useState([] as SavedQuery[]);
+}: SavedQueryManagementListProps) => {
+  const { uiSettings, notifications } = useKibana<IUnifiedSearchPluginServices>().services;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [currentPageNumber, setCurrentPageNumber] = useState(0);
+  const [totalQueryCount, setTotalQueryCount] = useState(0);
+  const [currentPageQueries, setCurrentPageQueries] = useState<SavedQuery[]>([]);
+  const [isInitializing, setIsInitializing] = useState(true);
+  const currentPageFetchId = useRef(0);
+  const selectableRef = useRef<EuiSelectable | null>(null);
   const [selectedSavedQuery, setSelectedSavedQuery] = useState(loadedSavedQuery);
-  const [toBeDeletedSavedQuery, setToBeDeletedSavedQuery] = useState(null as SavedQuery | null);
+  const [toBeDeletedSavedQuery, setToBeDeletedSavedQuery] = useState<SavedQuery | null>(null);
   const [showDeletionConfirmationModal, setShowDeletionConfirmationModal] = useState(false);
-  const cancelPendingListingRequest = useRef<() => void>(() => {});
-  const { uiSettings, http, application } = kibana.services;
   const format = uiSettings.get('dateFormat');
 
-  useEffect(() => {
-    const fetchCountAndSavedQueries = async () => {
-      cancelPendingListingRequest.current();
-      let requestGotCancelled = false;
-      cancelPendingListingRequest.current = () => {
-        requestGotCancelled = true;
-      };
+  const debouncedSetSearchTerm = useMemo(() => {
+    return debounce((newSearchTerm: string) => {
+      setSearchTerm((currentSearchTerm) => {
+        if (currentSearchTerm !== newSearchTerm) {
+          setCurrentPageNumber(0);
+        }
 
-      const savedQueryItems = await savedQueryService.getAllSavedQueries();
+        return newSearchTerm;
+      });
+    }, SAVED_QUERY_SEARCH_DEBOUNCE);
+  }, []);
 
-      if (requestGotCancelled) return;
+  const fetchPage = useLatest(async () => {
+    const fetchIdValue = ++currentPageFetchId.current;
+
+    try {
+      const preparedSearch = searchTerm.trim();
+      const { total, queries } = await savedQueryService.findSavedQueries(
+        preparedSearch || undefined,
+        SAVED_QUERY_PAGE_SIZE,
+        currentPageNumber + 1
+      );
+
+      if (fetchIdValue !== currentPageFetchId.current) {
+        return;
+      }
+
+      let filteredQueries = queries;
+
+      if (loadedSavedQuery && !preparedSearch && currentPageNumber === 0) {
+        filteredQueries = [
+          loadedSavedQuery,
```

Comment:
I don't think it's necessary/helpful to see the current query at the top of the list  if user intends to load another query. And it's weird that it makes it appear on more than 1 page.

![Feb-05-2024 11-02-26](https://github.com/elastic/kibana/assets/1415710/3efb0082-bf9f-44f5-88ca-8e1fb7c8eb7f)

Reply:
The main reason for this is an affordance for users to know which query is currently selected when they open the list, as well as providing quick access to delete it. Otherwise the query would be selected by default, but the list UI would give no indication to which query it was if it wasn't on the first page. It's a carryover from the old list (minus the showing up on multiple pages, which is a constraint due to the pagination), so I decided to keep the functionality for now.

That said, I think this could definitely be changed/improved, but I'd like to address it in a followup PR after this main one is merged. I'm going to create an issue including this as well as a few of the other UX issues like the "1 selected" label and the disabled title input for updates. That way we'll have some time to work with design on how to improve the UX, and address these issues in a followup PR.